### ISSUE TITLE
Problems with large cursors when scanning using RedisCluster 

### ISSUE THREAD

**Issue by bentleyo** [phpredis/phpredis#2600]
### Expected behaviour
Scans using `RedisCluster` that specify a large cursor should return results and update the cursor.

### Actual behaviour
`RedisClusterException` with the message "Couldn't process SCAN response from node" is thrown.

`RedisCluster` says the last error was "ERR invalid cursor".

### I'm seeing this behaviour on
- OS: Alpine Linux amd64 
- Redis: 7.1 (cluster mode and Amazon ElastiCache)
- PHP: 8.3.8
- phpredis: 6.1.0

### Steps to reproduce, backtrace or example script
Running the following PHP script:
```
try {
    $node = $redisCluster->_masters()[0];
    $it = null;
    do {
        var_dump($redisCluster->scan($it, $node, 'laravel_database_app-api-au-*', 20));
        var_dump($iterator);
    } while ($it !== 0);
} catch (\RedisClusterException $e) {
    var_dump($e);
}
```

The script returns the first scan correctly. The iterator is then updated to the string "9286422431638134784" which is expected from the changes in 6.1.0.

However, when the second scan is attempted the exception is thrown and that "ERR invalid cursor" message is happening.

To double check this isn't an issue with redis itself I attempted:
```
scan 0 match "laravel_database_app-api-au-*" count 20
```

Which returned the first page correctly and a similar cursor "9286422431638396928".
I then ran:
```
scan "9286422431638396928" match "laravel_database_app-api-au-*" count 20
```

It returned a new page of results and a new cursor value.

### I've checked
- [x ] There is no similar issue from other users
  - There's some discussion in #2562 but it's more about the initial cursor value (no longer works with the string `"0"`)
- [ ] Issue isn't fixed in `develop` branch

**Comment by bokanechase-digistorm** [comment]
@michael-grunder I noticed it looks like you implemented this feature in #2458 (thanks for that btw!). Warning, I'm not very familiar with C. However, I noticed that while `cursor` in `redis_cluster.c` is now type `uint64_t` ([see here](https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L2460)), when it gets passed into `redis_fmt_scan_cmd` as the parameter `it` it will have the type `long` ([see here](https://github.com/phpredis/phpredis/blob/develop/redis_commands.c#L576-L577)).

Could that be what's causing this to fail for me in this situation? I'm assuming part of the cursor is being lost before formatting.

**Comment by bentleyo** [comment]
@michael-grunder I noticed it looks like you implemented this feature in #2458 (side note - thanks for that!). Warning, I'm not very familiar with C. However, I noticed that while `cursor` in `redis_cluster.c` is now type `uint64_t` ([see here](https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L2460)), when it gets passed into `redis_fmt_scan_cmd` as the parameter `it` it will have the type `long` ([see here](https://github.com/phpredis/phpredis/blob/develop/redis_commands.c#L576-L577)).

Could that be what's causing this to fail for me in this situation? I'm assuming part of the cursor is being lost before formatting.
